lib/chef/knife/vsphere_vm_find.rb

```diff
@@ -0,0 +1,277 @@
+#
+# Author:: Raducu Deaconu (<rhadoo_io@yahoo.com>)
+# License:: Apache License, Version 2.0
+#
+
+require 'chef/knife'
+require 'chef/knife/base_vsphere_command'
+require 'rbvmomi'
+require 'netaddr'
+
+PsOn = 'poweredOn'
+PsOff = 'poweredOff'
+PsSuspended = 'suspended'
+
+#find vms belonging to pool that match criteria, display specified fields
+class Chef::Knife::VsphereVmFind < Chef::Knife::BaseVsphereCommand
+  banner "knife vsphere vm find"
+
+  common_options
+option :pool,
+        long: '--pool pool',
+        short: '-h',
+        description: 'Target pool'
+
+option :esx_disk,
+       long: '--esx-disk',
+       description: 'Show esx disks'
+
+option :snapshots,
+       long: '--snapshots',
+       description: 'Show snapshots'
+
+option :os_disk,
+       long: '--os-disks',
+       description: 'Show os disks'
+
+option :cpu,
+       long: '--cpu',
+       description: 'Show cpu'
+
+option :ram,
+       long: '--ram',
+       description: 'Show ram'
+
+option :ip,
+       long: '--ip',
+       description: 'Show ip'
+
+option :soff,
+       long: '--powered-off',
+       description: 'Show only stopped machines'
+       
+option :son,
+       long: '--powered-on',
+       description: 'Show only started machines'
+       
+option :matchip,
+       long: '--match-ip IP',
+       description: 'match ip'
+       
+option :matchos,
+       long: '--match-os OS',
+       description: 'match os'
+
+
+option :matchname,
+       long: '--match-name VMNAME',
+       description: 'match name'
+
+
+option :hostname,
+       long: '--hostname',
+       description: 'show hostname'
+
+option :os,
+       long: '--os',
+       description: 'show os details'
+
+option :alarms,
+       long: '--alarms',
+       description: 'show alarm status'
+
+option :tools,
+       long: '--tools',
+       description: 'show tools status'
+
+option :matchtools,
+       long: '--match-tools TOOLSSTATE',
+       description: 'match tools state'
+
+option :full_path,
+       long: '--full-path',
+       description: 'Show full path'
+
+
+
+$stdout.sync = true   #smoother output from print
+
+
+  def traverse_folders_for_pool(folder, poolname)
+    #children = folder.children.find_all
+    children = find_all_in_folder(folder, RbVmomi::VIM::ManagedObject)
```

Comment:
We've got this method in a couple other classes which signals to me that we should extract it to the base class (don't worry about that now). However this version is slightly different in the above two lines. What are you filtering out by looking for `RbVmomi::VIM::ManagedObject` that's coming back in `#find_all`? If it's an important distinction, what would you think about renaming this method so that it can't be confused with the other methods using the same name?